Objective-C/LogSink/CBLLogSinks.mm

```diff
@@ -0,0 +1,211 @@
+    
+    CBLLogLevel callbackLevel = std::min(consoleLevel, customLevel);
+    CBLLogLevel domainsLevel = std::min(callbackLevel, fileLevel);
+    C4LogLevel c4Level = (C4LogLevel)domainsLevel;
+    
+    if (_domainsLevel != domainsLevel) {
+        for (NSString* domain in c4Domains) {
+            C4LogDomain c4domain = c4log_getDomain([domain UTF8String], false);
+            if (c4domain)
+                c4log_setLevel(c4domain, c4Level);
+        }
+        
+        for (NSString* domain in platformDomains) {
+            C4LogDomain c4domain = c4log_getDomain([domain UTF8String], false);
+            if (c4domain)
+                c4log_setLevel(c4domain, c4Level);
+        }
+        _domainsLevel = domainsLevel;
+    }
+    
+    if (_callbackLevel != callbackLevel) {
+        c4log_writeToCallback(c4Level, &c4Callback, true);
+        _callbackLevel = callbackLevel;
+    }
+}
+
+static void c4Callback(C4LogDomain c4domain, C4LogLevel c4level, const char *msg, va_list args) {
+    NSString* message = [NSString stringWithUTF8String: msg];
+    CBLLogLevel level = (CBLLogLevel) c4level;
+    CBLLogDomain domain = toCBLLogDomain(c4domain);
+    [CBLLogSinks.console writeLogWithLevel: level domain: domain message :message];
+    [CBLLogSinks.custom writeLogWithLevel: level domain: domain message :message];
+}
+
++ (void) writeCBLLog: (C4LogDomain)c4domain level: (C4LogLevel)c4level message: (NSString*)message {
+    CBLLogLevel level = (CBLLogLevel) c4level;
+    CBLLogDomain domain = toCBLLogDomain(c4domain);
+    
+    CBLStringBytes c4msg(message);
+    c4slog(c4domain, c4level, c4msg);
+    
+    [CBLLogSinks.console writeLogWithLevel: level domain: domain message :message];
+    [CBLLogSinks.custom writeLogWithLevel: level domain: domain message :message];
+}
+
+
+static CBLLogDomain toCBLLogDomain(C4LogDomain domain) {
+    if (!domainDictionary) {
+        domainDictionary = @{ @"DB": @(kCBLLogDomainDatabase),
+                              @"Query": @(kCBLLogDomainQuery),
+                              @"Sync": @(kCBLLogDomainReplicator),
+                              @"SyncBusy": @(kCBLLogDomainReplicator),
+                              @"Changes": @(kCBLLogDomainDatabase),
+                              @"BLIP": @(kCBLLogDomainNetwork),
+                              @"WS": @(kCBLLogDomainNetwork),
+                              @"BLIPMessages": @(kCBLLogDomainNetwork),
+                              @"Zip": @(kCBLLogDomainNetwork),
+                              @"TLS": @(kCBLLogDomainNetwork),
+#ifdef COUCHBASE_ENTERPRISE
+                              @"Listener": @(kCBLLogDomainListener)
+#endif
+        };
+    }
+    
+    NSString* domainName = [NSString stringWithUTF8String: c4log_getDomainName(domain)];
+    NSNumber* mapped = [domainDictionary objectForKey: domainName];
+    return mapped ? mapped.integerValue : kCBLLogDomainDatabase;
+}
+
++ (void)checkLogApiVersion: (LogAPI)version {
+    if (!isInitialized)
```

Comment:
Not sure if it's required to be checked? Is there a case that the initialize() is not called and what will happen if not checking at all?

Reply:
needed to bypass 1 time (every test instance, at init) in order to be able to run both old and new tests. Should I just disable old tests (and remove this workaround)?

Reply:
I don't understand how this works. I don't see any code reset isInitialized. I think reset _vAPI to none would be enough for testing both old and new API?

Reply:
After reading the code, I have noticed that the isInitialized is reset to NO in CBLLogSinks's initialize. I couldn't find where that method is called in this PR. Also I think a better way is to have an internal method that the test can used to reset the API version.